src/dispatch/plugins/dispatch_slack/case/interactive.py

```diff
@@ -65,6 +73,87 @@ def assignee_select(
     )
 
 
+@message_dispatcher.add(
+    exclude={"subtype": ["channel_join", "channel_leave"]}
+)  # we ignore channel join and leave messages
+def handle_new_participant_added(ack: Ack, context: BoltContext, client: WebClient) -> None:
+    """Looks for new participants being added to conversation via @<user-name>"""
+    ack()
+    if context["subject"].type == "case":
+        participants = re.findall(context["message"], r"\<\@([a-zA-Z0-9]*)\>")
+        for user_id in participants:
+            user_email = get_user_email(client=client, user_id=user_id)
+
+            participant = case_flows.case_add_or_reactivate_participant_flow(
+                case_id=context["subject"].id, user_email=user_email
+            )
+            participant.user_conversation_id = user_id
+
+
+@message_dispatcher.add(
```

Comment:
afik we don't use `user_middleware` in cases because we don't have command scoped permissions. If we want to add permissions to modals we may need to add those in the future though.